shell/platform/windows/client_wrapper/include/flutter/flutter_engine.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_ENGINE_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_ENGINE_H_
+
+#define NOMINMAX
+#include <flutter_windows.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "dart_project.h"
+#include "plugin_registrar.h"
+#include "plugin_registry.h"
+
+namespace flutter {
+
+// An engine for running a headless Flutter application.
```

Comment:
Since flutter is GUI framework I'm a little confused what a headless Flutter application is and why you need a special engine for that scenario.  Might be worth expanding this comment a bit

Reply:
I added a bit more context about the current state to the comment.

Headless engines are a very niche (other than some add-to-app prewarming scenarios on mobile I believe, but we don't support that). The more important thing is that this lays the foundation for actually separating the view and the engine in the wrapper API; in the future things like running the loop will be done via the engine object, not the view controller. However, there are messy create/destroy semantics I need to figure out for that given that this isn't refcounted and needs to be somewhat coherent at both the wrapper and the C layers, so I didn't want to try to do that in this PR.